core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala

```diff
@@ -175,23 +176,23 @@ class ProducerFailureHandlingTest extends KafkaServerTestHarness {
   }
 
   /**
-   *  The send call with invalid partition id should throw KafkaException caused by IllegalArgumentException
-   */
+    * Send with invalid partition id should throw KafkaException when partition is higher than the
+    * upper bound of partitions and IllegalArgumentException when partition is negative
```

Comment:
I'll revisit in the context of `ProducerTest` (where the successful ones reside). One that I added and uses mocking around fetch is in `KafkaProducerTest`